File: haskell/series/src/Series.hs
module Series (slices) where

slices :: Int -> String -> [[Int]]
slices 0 _ = [[]]
slices n xs
    | isInvalid xs = []
    | otherwise = slices' n ints []
    where
        ints = map (read . (:"")) xs :: [Int]

slices' n [] result = result
slices' n ints@(x:xs) result
    | length ints > n = slices' n xs (result ++ [take n ints])
    | otherwise = result ++ [ints]

isInvalid xs = not $ all (`elem` ['0'..'9']) xs

File: python/series/series.py
def slices(series, span):
    length = len(series)
    if length < span or span == 0:
        raise ValueError
    else:
        out = []
        ints = [int(i) for i in series]
        return [ints[idx:span+idx] for idx in range(length + 1 - span)]
